pallets/msa/src/lib.rs

```diff
@@ -459,6 +466,26 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Current user can retire their own MSA, so that it cannot be used again.
+		/// Returns `Ok(())` on success, otherwise returns an error. Deposits event [`MSARetired`](Event::MSARetired).
+		///
+		/// ### Errors
+		/// - Returns ['NotMsaOwner'](Error::NotMsaOwner)
+		#[pallet::weight(T::WeightInfo::retire_my_msa())]
+		pub fn retire_my_msa(origin: OriginFor<T>) -> DispatchResult {
+
+			// fetch msa id from origin
+			let account_id = ensure_signed(origin)?;
+			let msa_id = Self::ensure_valid_msa_key(&account_id)?.msa_id.into();
+
+			// call self::retire_msa()
```

Comment:
I don't think you need these comments since the functionality and names are self explanatory